testing/test_config.py

```diff
@@ -843,3 +843,10 @@ def test_with_existing_file_in_subdir(self, tmpdir):
             rootdir, inifile, inicfg = determine_setup(None, ['a/exist'])
             assert rootdir == tmpdir
             assert inifile is None
+
+    def test_addopts_before_initini(self, testdir, tmpdir, monkeypatch):
+        monkeypatch.setenv('PYTEST_ADDOPTS', '-o cache_dir=/somewhere')
```

Comment:
good thinking, for more pedantic separation i propose to replace the arbitrarily chosen `/somewhere` by using `testdir.join('somewhere')`

Reply:
Done in 48c31f9bd7f08d4cd30e5396503b0899668452c7. I had to use `testdir.tmpdir.join(...)` though.